haproxy/entrypoint.sh

```diff
@@ -1,30 +1,46 @@
 #!/bin/bash
-
-set -e
+set -euo pipefail
 
 DEFAULT="/usr/local/etc/haproxy/default_frontend.cfg"
 BACKEND="/usr/local/etc/haproxy/backend.cfg"
+PASSWD_DIR="/srv/storage/haproxy/passwd"
 
 # Update backend servers
-cp /usr/local/etc/haproxy/backend.cfg.template $BACKEND
-# shellcheck disable=SC2153
-for COUCHDB_SERVER in ${COUCHDB_SERVERS//,/ }
-do
-  server="  server $COUCHDB_SERVER $COUCHDB_SERVER:5984 check agent-check agent-inter 5s agent-addr $HEALTHCHECK_ADDR agent-port 5555"
-  if [[ -n "${DOCKER_DNS_RESOLVER:-}" ]]; then
-    server="${server} resolvers docker_resolver resolve-prefer ipv4"
-  fi
-  echo "${server}" >> $BACKEND
-done
+cp "${BACKEND}.template" "$BACKEND"
+
+# Read servers into array, trim whitespace
+readarray -td, SERVERS < <(echo -n "${COUCHDB_SERVERS}" | tr -d '[:space:]')
 
-# Place environment variables into config
-envsubst < $DEFAULT
-envsubst < $BACKEND
+setResolver() {
+    local server="$1"
+    if [[ -n "${DOCKER_DNS_RESOLVER:-}" ]]; then
+        server="${server} resolvers docker_resolver resolve-prefer ipv4"
+    fi
+    echo "$server"
+}
 
-#Write pw for healthcheck subshell to work
-mkdir -p /srv/storage/haproxy/passwd
-echo "$COUCHDB_USER" > /srv/storage/haproxy/passwd/username
-echo "$COUCHDB_PASSWORD" > /srv/storage/haproxy/passwd/admin
+if [[ ${#SERVERS[@]} -eq 1 ]]; then
+    {
+      basic_auth=$(echo -n "$COUCHDB_USER:$COUCHDB_PASSWORD" | base64)
+      echo "  option httpchk"
+      echo "  http-check send meth GET uri /_up hdr Authorization 'Basic ${basic_auth}'" >> "$BACKEND"
+      echo "  http-check expect status 200" >> "$BACKEND"
+      config="  server ${SERVERS[0]} ${SERVERS[0]}:5984 check inter 5s"
+      setResolver "$config"
+    } >> "$BACKEND"
+else
+    for server in "${SERVERS[@]}"; do
+        [[ -z "$server" ]] && continue  # Skip empty entries
+        config="  server ${server} ${server}:5984 check agent-check agent-inter 5s agent-addr ${HEALTHCHECK_ADDR:-localhost} agent-port 5555"
```

Comment:
`${HEALTHCHECK_ADDR:-localhost}` is the default to `localhost` here intentional/necessary?  We always need to be calling out to the healthcheck container here, right?

Reply:
Yes, we always need to call out healthcheck here, and we should _always_ need to have  `HEALTHCHECK_ADDR` set, I think the `localhost` was probably used for testing at some point, or when network is set to be host mode in docker. 

I think that if we remove the default value and not set HEALTHCHECK_ADDR, we would get some weird config error in haproxy, and if we keep the value then we would get a better error that checking the server failed (because nothing runs on port 5555 on haproxy host). 
Maybe the best is to always require HEALTHCHECK_ADDR to be set. 

Honestly I just took this code from the previous version and didn't think about it at all.